tests/src/core/testqgsexpression.cpp

```diff
@@ -821,6 +821,8 @@ class TestQgsExpression: public QObject
       QTest::newRow( "upper" ) << "upper('HeLLo')" << false << QVariant( "HELLO" );
       QTest::newRow( "length" ) << "length('HeLLo')" << false << QVariant( 5 );
       QTest::newRow( "replace" ) << "replace('HeLLo', 'LL', 'xx')" << false << QVariant( "Hexxo" );
+      QTest::newRow( "replace_array (array replaced by array)" ) << "replace_array('321', array('1','2','3'), array('7','8','9'))" << false << QVariant( "987" );
```

Comment:
Can you add tests with unbalanced array lengths? Ie one with before > after, and vice versa.